rskj-core/src/main/java/co/rsk/db/importer/BootstrapImporter.java

```diff
@@ -102,10 +81,58 @@ private static void insertBlocks(BlockStore blockStore,
             RLPList blockData = RLP.decodeList(element.getRLPData());
             RLPList tuple = RLP.decodeList(blockData.getRLPData());
             Block block = blockFactory.decodeBlock(tuple.get(0).getRLPData());
-            BlockDifficulty blockDifficulty = new BlockDifficulty(new BigInteger(tuple.get(1).getRLPData()));
+            BlockDifficulty blockDifficulty = new BlockDifficulty(new BigInteger(Objects.requireNonNull(tuple.get(1).getRLPData())));
             blockStore.saveBlock(block, blockDifficulty, true);
         }
 
         blockStore.flush();
     }
+
+    private static void prepareState(Queue<byte[]> nodeDataQueue, HashMapDB hashMapDB, RLPElement rlpElement) {
+        Queue<RLPElement> nodeListQueue = decodeQueue(rlpElement.getRLPData());
+
+        prepareNodeData(nodeDataQueue, hashMapDB, RLP.decodeList(Objects.requireNonNull(nodeListQueue.poll()).getRLPData()));
+        prepareNodeValues(hashMapDB, RLP.decodeList(Objects.requireNonNull(nodeListQueue.poll()).getRLPData()));
+    }
+
+    private static void prepareNodeData(Queue<byte[]> rlpDataQueue, HashMapDB hashMapDB, RLPList nodesData) {
+        int size = nodesData.size();
+        for (int k = 0; k < size; k++) {
+            RLPElement element = nodesData.get(k);
+            byte[] rlpData = Objects.requireNonNull(element.getRLPData());
+
+            hashMapDB.put(Keccak256Helper.keccak256(rlpData), rlpData);
```

Comment:
Previously, the `fakeDB` had as keys the `.getHash()` of the trie, which passes through quite a bit of serialization and changes, instead of directly hashing the `rlpData`. Are they equivalent somehow? I may be missing something but I think this could mess up other methods such as `TrieStoreImpl.save()`, as `.getHash()` of the fakeStore is used there...

TL;DR: The fake store and the _actual_ store use different keys, which may end up impacting the actual store as the actual store relies on the fake store `getHash` method.